src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -89,4 +89,12 @@ public void ApplyChanges(MicrobeEditor editor)
         storedOrganelle.SetCustomUpgradeObject(new ChemoreceptorUpgrades(shownChoices[compounds.Selected],
             (float)maximumDistance.Value, (float)minimumAmount.Value, colour.Color));
     }
+
+    public void CompoundChanged(int index)
+    {
+        if (shownChoices != null)
+        {
+            colour.Color = shownChoices[index].Colour; // when the color changes then it changes it to match the color
```

Comment:
Using trailing comments isn't really the Thrive style so you should put this comment on a line by itself (and start it with a capital letter).

Also there's one part of the issue that's left unimplemented here:
> Could maybe only change the colour if it is white or one of the other compound's colour. That would make it less annoying for players as if they first selected a custom colour and then selected a different compound, they'd lose that colour without that check.

So this should first check if the current colour is *any* of the compound colours and only *then* update the colour, so that if the player has selected a custom colour before changing the compound type, they don't lose that selection.